cmd/state-svc/main.go

```diff
@@ -192,6 +192,15 @@ func runForeground(cfg *config.Instance, an *anaSync.Client, auth *authenticatio
 	signal.Notify(sig, os.Interrupt, syscall.SIGTERM)
 	defer signal.Stop(sig)
 
+	p.RunIfNotAuthority(time.Second*3, svcctl.NewDefaultIPCClient(), func(err error) {
+		fmt.Fprintln(os.Stderr, err)
```

Comment:
Debugging code? Not sure I understand why this one is printed but the Stop error is logged.

Reply:
This error is printed because it isn't exactly expected by the user, but I didn't want to return non-zero. It is being printed just as the main function prints errors there: 4f636e9969b885b7241d139d9ffe38b9cc7c983da7882b66dcee6211b9ba88b0L82

Reply:
Ok, but why not return it? `runForeground()` returns errors, and its caller will print them. Printing them here doesn't seem desirable.

Reply:
Returning an error would mean that we have to wait for it, and if the application dies before the check start or completes, I think we should be ok with not reaching the error. More moving parts can be added to make RunIfNotAuthority more mechanically correct, but I didn't think it was worthwhile.

Reply:
I think I worded my previous message poorly. RunIfNotAuthority is running concurrently. If we block on it, then the error returned from Wait may never be reached. Specifically, if the server has a problem starting up, we want to response to that immediately and not have to wait for a later scheduled authority check. We could wait, and collapse the scheduled authority check, but that would require a more complex mechanism. Printing the stop failure is the simplest solution that permits us to know of a Stop failure and still reports a Wait failure.

Reply:
I see. That makes sense, thanks for sticking with it.